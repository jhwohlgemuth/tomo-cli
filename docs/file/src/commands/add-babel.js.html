<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/commands/add-babel.js | tomo-cli</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="A friendly command line tool designed to help create sustainable software using web technology"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="tomo-cli"><meta property="twitter:description" content="A friendly command line tool designed to help create sustainable software using web technology"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a style="position:relative; top:3px;" href="https://github.com/jhwohlgemuth/tomo-cli"><img width="20px" src="./image/github.png"></a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CommandError">CommandError</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OfflineWarning">OfflineWarning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Task">Task</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TaskList">TaskList</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-UI">UI</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Warning">Warning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-populateQueue">populateQueue</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/BasicEditor.js~BasicEditor.html">BasicEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/MakefileEditor.js~MakefileEditor.html">MakefileEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/Scaffolder.js~Scaffolder.html">Scaffolder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExist">allDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExistSync">allDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExist">allDoNotExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExistSync">allDoNotExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-format">format</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getBinDirectory">getBinDirectory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getCommandDirectory">getCommandDirectory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-parse">parse</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExist">someDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExistSync">someDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createJsonEditor">createJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createModuleEditor">createModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getIntendedInput">getIntendedInput</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getVersions">getVersions</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-install">install</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-verifyRustInstallation">verifyRustInstallation</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-BabelConfigModuleEditor">BabelConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-EslintConfigModuleEditor">EslintConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PackageJsonEditor">PackageJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PostcssConfigEditor">PostcssConfigEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-WebpackConfigEditor">WebpackConfigEditor</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/commands/add-babel.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import {BabelConfigModuleEditor, install, PackageJsonEditor} from &apos;../utils&apos;;
import {allDoNotExist, someDoExist} from &apos;../utils/common&apos;;

const BABEL_CORE = [
    &apos;@babel/cli&apos;,
    &apos;@babel/core&apos;,
    &apos;@babel/runtime&apos;
];
const BABEL_PRESETS = [
    &apos;@babel/preset-env&apos;
];
const BABEL_PLUGINS = [
    &apos;@babel/plugin-transform-runtime&apos;,
    &apos;@babel/plugin-proposal-class-properties&apos;,
    &apos;@babel/plugin-proposal-export-default-from&apos;,
    &apos;@babel/plugin-proposal-optional-chaining&apos;
];
const BABEL_REACT_PRESET = [
    &apos;@babel/preset-react&apos;
];
const BABEL_DEPENDENCIES = [
    ...BABEL_CORE,
    ...BABEL_PRESETS,
    ...BABEL_PLUGINS
];
/** @ignore */
export const tasks = [
    {
        text: &apos;Create Babel config file&apos;,
        task: async () =&gt; {
            await (new BabelConfigModuleEditor()).create().commit();
        },
        condition: () =&gt; allDoNotExist(&apos;babel.config.js&apos;, &apos;.babelrc&apos;, &apos;.babelrc.js&apos;)
    },
    {
        text: &apos;Install Babel core, CLI, presets, and plugins&apos;,
        task: ({skipInstall}) =&gt; install(BABEL_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) =&gt; isNotOffline &amp;&amp; (!(new PackageJsonEditor()).hasAll(...BABEL_DEPENDENCIES) &amp;&amp; someDoExist(&apos;package.json&apos;))
    },
    {
        text: &apos;Install Babel React preset&apos;,
        task: ({skipInstall}) =&gt; install(BABEL_REACT_PRESET, {dev: true, skipInstall}),
        condition: ({useReact}) =&gt; (useReact &amp;&amp; someDoExist(&apos;package.json&apos;)),
        optional: ({useReact}) =&gt; useReact
    },
    {
        text: &apos;Add React support to Babel configuration file&apos;,
        task: async () =&gt; {
            const presets = [...BABEL_PRESETS, ...BABEL_REACT_PRESET];
            await (new BabelConfigModuleEditor()).extend({presets}).commit();
        },
        condition: ({useReact}) =&gt; (useReact &amp;&amp; someDoExist(&apos;babel.config.js&apos;)),
        optional: ({useReact}) =&gt; useReact
    }
];
export default tasks;</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
