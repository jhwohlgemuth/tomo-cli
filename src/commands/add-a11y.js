import {
    PackageJsonEditor,
    allDoExist,
    install,
    uninstall
} from '../api';
/**
 * @type {task[]}
 * @see http://pa11y.org/
 */
export const addA11y = [
    {
        text: 'Add accessibility tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                'lint:aria': `pa11y ${outputDirectory}/index.html`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install pa11y for checking accessibility rules',
        task: ({skipInstall}) => install(['pa11y'], {dev: true, skipInstall}),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json')
    }
];
export const removeA11y = [
    {
        text: 'Remove accessibility tasks from package.json',
        task: async () => {
            const scripts = {
                'lint:aria': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall pa11y',
        task: () => uninstall(['pa11y']),
        condition: () => allDoExist('package.json') && (new PackageJsonEditor()).hasAll('pa11y')
    }
];
export default addA11y;