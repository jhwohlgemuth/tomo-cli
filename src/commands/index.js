/* eslint-disable no-magic-numbers */
import {choose, withOptions} from '../utils';
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import {addBrowsersync} from './add-browsersync';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMakefile from './add-makefile';
import addMarionette from './add-marionette';
import {addParcel, removeParcel} from './add-parcel';
import {addPostcss, removePostcss} from './add-postcss';
import {addRollup, removeRollup} from './add-rollup';
import {addWebpack, removeWebpack} from './add-webpack';
import createServer from './create-server';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addPostcss,
        ...addJest,
        choose({
            default: addMarionette,
            useReact: []// do nothing
        }),
        choose({
            default: addWebpack,
            useRollup: addRollup,
            useParcel: addParcel
        }),
        choose({
            default: addBrowsersync,
            useParcel: []// do nothing
        })
    ],
    server: [
        withOptions({sourceDirectory: '.'}),
        ...createPackageJson,
        ...addEslint,
        ...createServer
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    browsersync: addBrowsersync,
    esdoc: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    makefile: addMakefile,
    parcel: [
        ...addBabel,
        ...addParcel
    ],
    postcss: addPostcss,
    rollup: [
        ...addBabel,
        ...addRollup
    ],
    webpack: [
        ...addBabel,
        ...addWebpack
    ]
};
const remove = {
    parcel: removeParcel,
    postcss: removePostcss,
    rollup: removeRollup,
    webpack: removeWebpack
};

export default {
    add,
    remove,
    create,
    new: create// alias for create
};
/**
 * @typedef {Object} task
 * @property {string} text Display text for task
 * @property {function} task Task to execute
 * @property {function} condition Predicate to decide when to execute the task (true) or not (false) - can be async or sync
 * @property {function} [optional] Predicate to decide when to show the task (true) or not (false) - MUST be sync
 */