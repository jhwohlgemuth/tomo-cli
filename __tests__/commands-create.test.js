import {fileContents, getDirectoryTree, run, useTemporaryDirectory} from './tomo-test';
import commands from '../src/commands';

jest.mock('is-online', () => (async () => true));

describeOnlyOnLinux('Create', () => {
    let tempDirectory;
    const skipInstall = true;
    const useReact = true;
    const withRust = true;
    const reactVersion = '16.2';
    const {create} = commands;
    const omit = ['extension', 'path', 'size', 'type'];
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('new project', async () => {
        await run(create.project, {skipInstall});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const config = fileContents('./.editorconfig');
        expect(tree).toMatchSnapshot();
        expect(config).toMatchSnapshot();
    });
    test('new app', async () => {
        await run(create.app, {skipInstall});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('./package.json');
        const cfg = fileContents('./.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
    test('new react app', async () => {
        const options = {reactVersion, skipInstall, useReact};
        await run(create.app, options);
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('./package.json');
        const cfg = fileContents('./.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
    test('new react app (with Rust)', async () => {
        const options = {reactVersion, skipInstall, useReact, withRust};
        await run(create.app, options);
        const tree = getDirectoryTree(tempDirectory, {omit});
        expect(tree).toMatchSnapshot();
    });
    test('new server', async () => {
        await run(create.server, {skipInstall});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('./package.json');
        const cfg = fileContents('./.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
});