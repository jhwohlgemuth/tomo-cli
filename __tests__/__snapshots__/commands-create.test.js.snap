// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create new app 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .editorconfig
        },
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: .gitignore
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js
                },
                {
                    name: setup.js
                }
            ]
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: fonts.css
                        },
                        {
                            name: style.css
                        }
                    ]
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: SansForgetica-Regular.eot
                        },
                        {
                            name: SansForgetica-Regular.svg
                        },
                        {
                            name: SansForgetica-Regular.ttf
                        },
                        {
                            name: SansForgetica-Regular.woff
                        },
                        {
                            name: SansForgetica-Regular.woff2
                        }
                    ]
                },
                {
                    name: images,
                    children: [
                        {
                            name: blank_canvas.png
                        },
                        {
                            name: preferences.png
                        }
                    ]
                },
                {
                    name: index.html
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                }
            ]
        },
        {
            name: babel.config.js
        },
        {
            name: package.json
        },
        {
            name: postcss.config.js
        },
        {
            name: source,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: app.js
                        },
                        {
                            name: body.js
                        },
                        {
                            name: footer.js
                        },
                        {
                            name: header.js
                        }
                    ]
                },
                {
                    name: main.js
                },
                {
                    name: plugins,
                    children: [
                        {
                            name: mn.radio.logging.js
                        },
                        {
                            name: mn.redux.state.js
                        }
                    ]
                },
                {
                    name: shims,
                    children: [
                        {
                            name: mn.renderer.shim.js
                        }
                    ]
                }
            ]
        },
        {
            name: webpack.config.js
        }
    ]
}
"
`;

exports[`Create new app 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"build:es\\": \\"webpack\\",
        \\"watch:es\\": \\"watch \\\\\\"npm run build:es\\\\\\" ./source\\",
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch 'npm run lint' ./source\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./output\\",
        \\"watch:css\\": \\"npm run build:css -- --watch\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:ing\\": \\"npm test -- --watchAll\\",
        \\"predeploy\\": \\"npm-run-all clean \\\\\\"build:es -- --mode=production\\\\\\" build:css\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:stats\\": \\"webpack --mode production --profile --json > stats.json\\",
        \\"build:analyze\\": \\"webpack-bundle-analyzer ./stats.json\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"dashboard\\": \\"webpack-dashboard -- webpack-dev-server --config ./webpack.config.js\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run dashboard\\\\\\" : \\\\\\"npm run lint:ing\\\\\\" ]\\",
        \\"prestart\\": \\"npm run build:es\\",
        \\"start\\": \\"npm-run-all --parallel watch:es watch:css serve\\",
        \\"serve\\": \\"browser-sync start --server ./output --files ./output --port 4669\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"setupFilesAfterEnv\\": [
            \\"<rootDir>/__tests__/setup.js\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"./source/main.js\\"
}
"
`;

exports[`Create new app 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:lit/recommended'
    ],
    parser: 'babel-eslint',
    plugins: [
        'lit'
    ]
};"
`;

exports[`Create new project 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .editorconfig
        },
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: babel.config.js
        },
        {
            name: package.json
        }
    ]
}
"
`;

exports[`Create new project 2`] = `
"[*]
indent_style = space

[*.js]
charset = utf-8
end_of_line = lf
indent_size = 4
insert_final_newline = false
max_line_length = 150
trim_trailing_whitespace = true

[*.json]
indent_size = 2"
`;

exports[`Create new react app (with Rust) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .editorconfig
        },
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: .gitignore
        },
        {
            name: Cargo.toml
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js
                },
                {
                    name: setup.js
                }
            ]
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: fonts.css
                        },
                        {
                            name: style.css
                        }
                    ]
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: SansForgetica-Regular.eot
                        },
                        {
                            name: SansForgetica-Regular.svg
                        },
                        {
                            name: SansForgetica-Regular.ttf
                        },
                        {
                            name: SansForgetica-Regular.woff
                        },
                        {
                            name: SansForgetica-Regular.woff2
                        }
                    ]
                },
                {
                    name: images,
                    children: [
                        {
                            name: preferences.png
                        },
                        {
                            name: react.png
                        }
                    ]
                },
                {
                    name: index.html
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                }
            ]
        },
        {
            name: babel.config.js
        },
        {
            name: package.json
        },
        {
            name: postcss.config.js
        },
        {
            name: rust-to-wasm,
            children: [
                {
                    name: Cargo.toml
                },
                {
                    name: src,
                    children: [
                        {
                            name: lib.rs
                        },
                        {
                            name: utils.rs
                        }
                    ]
                },
                {
                    name: tests,
                    children: [
                        {
                            name: app.rs
                        },
                        {
                            name: web.rs
                        }
                    ]
                }
            ]
        },
        {
            name: source,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: App.jsx
                        },
                        {
                            name: Body.jsx
                        },
                        {
                            name: Footer.jsx
                        },
                        {
                            name: Header.jsx
                        }
                    ]
                },
                {
                    name: main.jsx
                }
            ]
        },
        {
            name: webpack.config.js
        }
    ]
}
"
`;

exports[`Create new react app 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .editorconfig
        },
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: .gitignore
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js
                },
                {
                    name: setup.js
                }
            ]
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: fonts.css
                        },
                        {
                            name: style.css
                        }
                    ]
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: SansForgetica-Regular.eot
                        },
                        {
                            name: SansForgetica-Regular.svg
                        },
                        {
                            name: SansForgetica-Regular.ttf
                        },
                        {
                            name: SansForgetica-Regular.woff
                        },
                        {
                            name: SansForgetica-Regular.woff2
                        }
                    ]
                },
                {
                    name: images,
                    children: [
                        {
                            name: preferences.png
                        },
                        {
                            name: react.png
                        }
                    ]
                },
                {
                    name: index.html
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep
                        }
                    ]
                }
            ]
        },
        {
            name: babel.config.js
        },
        {
            name: package.json
        },
        {
            name: postcss.config.js
        },
        {
            name: source,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: App.jsx
                        },
                        {
                            name: Body.jsx
                        },
                        {
                            name: Footer.jsx
                        },
                        {
                            name: Header.jsx
                        }
                    ]
                },
                {
                    name: main.jsx
                }
            ]
        },
        {
            name: webpack.config.js
        }
    ]
}
"
`;

exports[`Create new react app 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"build:es\\": \\"webpack\\",
        \\"watch:es\\": \\"webpack-dev-server --hot --open --mode development\\",
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch 'npm run lint' ./source\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./output\\",
        \\"watch:css\\": \\"npm run build:css -- --watch\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:ing\\": \\"npm test -- --watchAll\\",
        \\"predeploy\\": \\"npm-run-all clean \\\\\\"build:es -- --mode=production\\\\\\" build:css\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:stats\\": \\"webpack --mode production --profile --json > stats.json\\",
        \\"build:analyze\\": \\"webpack-bundle-analyzer ./stats.json\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"dashboard\\": \\"webpack-dashboard -- webpack-dev-server --config ./webpack.config.js\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run dashboard\\\\\\" : \\\\\\"npm run lint:ing\\\\\\" ]\\",
        \\"start\\": \\"npm-run-all build:es --parallel watch:*\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"setupFilesAfterEnv\\": [
            \\"<rootDir>/__tests__/setup.js\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"./source/main.js\\"
}
"
`;

exports[`Create new react app 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:react/recommended',
        'plugin:jsx-a11y/recommended'
    ],
    parser: 'babel-eslint',
    plugins: [
        'jsx-a11y'
    ],
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    settings: {
        react: {
            version: '16.2'
        }
    }
};"
`;

exports[`Create new server 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .env
        },
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: .gitignore
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js
                }
            ]
        },
        {
            name: config,
            children: [
                {
                    name: default.js
                },
                {
                    name: test.js
                }
            ]
        },
        {
            name: db.json
        },
        {
            name: favicon.ico
        },
        {
            name: graphql.js
        },
        {
            name: index.js
        },
        {
            name: package.json
        },
        {
            name: public,
            children: [
                {
                    name: example.md
                },
                {
                    name: index.html
                },
                {
                    name: main.js
                }
            ]
        },
        {
            name: server.js
        },
        {
            name: socket.js
        },
        {
            name: ssl,
            children: [
                {
                    name: server.cert
                },
                {
                    name: server.key
                }
            ]
        }
    ]
}
"
`;

exports[`Create new server 2`] = `
"{
    \\"name\\": \\"tomo-server\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"Node.js HTTP(S), WebSocket, and GraphQL servers with an 80% solution for security 'baked in'\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch 'npm run lint' .\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:ing\\": \\"npm test -- --watchAll\\",
        \\"predev\\": \\"npm run open\\",
        \\"dev\\": \\"stmux [ \\\\\\"nodemon index.js\\\\\\" : \\\\\\"npm run lint:ing\\\\\\" ]\\",
        \\"prestart\\": \\"npm audit --production\\",
        \\"start\\": \\"node index.js\\",
        \\"open\\": \\"open-cli http://localhost:8111\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"index.js\\"
}
"
`;

exports[`Create new server 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: false,
        node: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:react/recommended',
        'plugin:jsx-a11y/recommended'
    ],
    parser: 'babel-eslint',
    plugins: [
        'jsx-a11y'
    ],
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    settings: {
        react: {
            version: '16.2'
        }
    }
};"
`;
